### PR TITLE
Add more unit tests for Core

### PR DESCRIPTION
**Pull request by srnagar**
Added some more unit tests for azure-core and removed some duplicate/unused code.

### CODE REVIEW

File: core/azure-core/src/main/java/com/azure/core/implementation/Validator.java [hunk 1]
```diff
@@ -40,16 +40,7 @@ public static void validate(Object parameter) {
...
-        if (type == Double.class
-                || type == Float.class
-                || type == Long.class
-                || type == Integer.class
-                || type == Short.class
-                || type == Character.class
-                || type == Byte.class
-                || type == Boolean.class) {
-            type = wrapperToPrimitive(type);
-        }
```
Comment (JonathanGiles) [inline core/azure-core/src/main/java/com/azure/core/implementation/Validator.java]: Why did you remove this? Are you sure it is never used?

File: core/azure-core/src/test/java/com/azure/core/implementation/util/FluxUtilTests.java [hunk 2]
```diff
@@ -246,6 +249,19 @@ public void toByteArrayWithNonEmptyByteBuffer() {
         assertArrayEquals(new byte[] { 0, 1, 2, 3, 4 }, byteArray);
         assertEquals(5, byteBuffer.readableBytes());
     }
+
+    @Test
+    public void testCollectBytBufStream() {
```
Comment (JonathanGiles) [inline core/azure-core/src/test/java/com/azure/core/implementation/util/FluxUtilTests.java]: s/BytBuf/ByteBuf

File: core/azure-core/src/main/java/com/azure/core/implementation/Validator.java (hunk 1 above)
Comment (srnagar) [reply core/azure-core/src/main/java/com/azure/core/implementation/Validator.java]: Yeah, the `wrapperToPrimitive` method did nothing but return the input `type`. Not sure why this was added but just to be sure, I added a test case to cover primitive types.

Comment (JonathanGiles) [reply core/azure-core/src/main/java/com/azure/core/implementation/Validator.java]: I don't believe this is accurate - wrapperToPrimitive returns, e.g. Integer.TYPE when given an Integer.class - these are two separate class types (the first represents the primitive, the second represents the reference type).

Comment (srnagar) [reply core/azure-core/src/main/java/com/azure/core/implementation/Validator.java]: If the type is Integer.class, the `wrapperToPrimitive` method has a check `if ( !clazz.isPrimitive() ) {return clazz;}` which will just return Integer.class because the `isPrimitive` check fails. So, it will not be able to return `Integer.TYPE`. Maybe that `isPrimitive` check is not required?

Comment (JonathanGiles) [reply core/azure-core/src/main/java/com/azure/core/implementation/Validator.java]: Yes, good point! I don't know why this code exists. I think it is probably safe to remove it if you've tested it enough.

Comment (srnagar) [reply core/azure-core/src/main/java/com/azure/core/implementation/Validator.java]: Yeah, I tested with all primitive and boxed types of int, short, long, byte, boolean, char, float and none of them went past the `!class.isPrimitive()` check. So, I think it's safe to remove it.
